### DISCUSSION TITLE
無法linkplay

### DISCUSSION THREAD

**Discussion by WhiteCloud0106** [Lost-MSth/Arcaea-server#147] [Q&A] [answered]
在客戶端按下創建房間的時候服務端正常創建，但是客戶端會在loading一段時間後顯示"無法連線至線上伺服器"
我直接用服務端顯示的六位房號加入房間則會顯示"此房間正在遊玩歌曲，目前無法加入"
服務端跑出以下訊息
![image](https://github.com/Lost-MSth/Arcaea-server/assets/94116682/63f45694-6c4d-4786-8af7-3b556b255851)

**Comment by Lost-MSth** [comment, marked as answer]
UDP 连接不通，如果是云服务器请开启 10900 UDP 端口

**Reply by WhiteCloud0106** [reply]
哈嘍，我們服的玩家發現了一些新東西
ios好像是連得上的，但安卓客戶端的狀況跟之前一樣。
但是如果安卓客戶端顯示“無法連線至線上伺服器”的話，ios的玩家其實可以看到該玩家有進去房間 只是無法動作。
我已經無法理解現在什麼狀況了hhh(我自己猜跟殼子有關？

**Reply by WhiteCloud0106** [reply]
剛剛他們又測試了一陣子，ios玩家是可以成功link play的

**Reply by Lost-MSth** [reply]
多半壳子问题，我并没有复现出来，模拟器、真机都试过了

**Reply by WhiteCloud0106** [reply]
好像確實是，剛剛問幫忙弄殼子的發現有些東西沒弄到
對了順便問一下服務器有一段時間沒有紀錄world rank的分數了
不知道是甚麼問題
這半年的新玩家不管打多少 database裡world rank用的分數那欄都是0

**Reply by Lost-MSth** [reply]
谱面有定数吗，只有有定数的 FTR 和 BYD 才会记入（尚不清楚官服机制）

**Reply by WhiteCloud0106** [reply]
都有的
雖然是從網頁端輸進去的
但是輸完都有重啟服務器

**Reply by Lost-MSth** [reply]
尚未复现。服务端有魔改过吗？

**Reply by WhiteCloud0106** [reply]
應該是沒有吧🤔
還是您講一下位置我去看有沒有問題

**Reply by Lost-MSth** [reply]
https://github.com/Lost-MSth/Arcaea-server/blob/2d498ae02b021f7fbb7e7f38866942bb3e48bd31/latest%20version/core/user.py#L664

https://github.com/Lost-MSth/Arcaea-server/blob/2d498ae02b021f7fbb7e7f38866942bb3e48bd31/latest%20version/core/score.py#L489

**Reply by WhiteCloud0106** [reply]
結果好像score.py的那個被註釋掉了，應該是我朋友不知道為什麼拿掉的
把註釋移除之後好像就可以了，感謝您！（如果還有問題我再繼續問（

**Comment by WhiteCloud0106** [comment]
![image](https://github.com/Lost-MSth/Arcaea-server/assets/94116682/ca92e3d0-77fe-44a1-8534-b0e4b8267575)
我下載了較新版本在我的電腦上，但啟動時報錯了，看不太懂甚麼原因

**Reply by Lost-MSth** [reply]
你在服务端 `songs` 文件夹下放了 `songlist` 文件，而且这个文件的格式是错误的，具体问题在于某个歌曲的 `additional_files` 不是 `list[dict]` 格式，你写的是 `list[str]` 格式

首先请参考最新的官方 `songlist` 文件来写，其次若无必须不要放 `songlist` 文件
